File: BOJ/[1092] 배/DDAGUE/1092.py
N = int(input())
crane = list(map(int, input().split()))
M = int(input())
box = list(map(int, input().split()))
time = 0
check = [0 for _ in range(M)]
cnt = 0

crane.sort(reverse = True)
box.sort(reverse = True)

place = [0] * N

if max(box) <= max(crane):
    while cnt < len(box):
        for i in range(N): # 크레인에 대하여
            while place[i] < len(box):
                if not check[place[i]] and crane[i] >= box[place[i]]:
                    check[place[i]] = True
                    place[i] += 1
                    cnt += 1
                    break
                place[i] += 1
        time += 1
    print(time)
else:
    print(-1)

File: BOJ/[1092] 배/pinkishincoloragain/1092.js
const fs = require("fs");
let input = fs
	// .readFileSync("./input.txt")
	.readFileSync("/dev/stdin")
	.toString()
	.trim()
	.split("\n");

const C = +input[0]; //Crane length
const B = +input[2]; //Box length
let crane = input[1].split(" ").map(Number);
let box = input[3].split(" ").map(Number);

crane.sort((a, b) => b - a);
box.sort((a, b) => b - a);

let time = 0;

if (crane[0] < box[0]) {
	console.log(-1);
} else {
	while (box.length) {
		let ci = 0;
		for (let bi = 0; bi < box.length; bi++) {
			if (ci === crane.length) break;
			if (crane[ci] >= box[bi]) {
				box.splice(bi, 1);
				bi--;
				ci++;
			}
		}
		time += 1;
	}

	console.log(time);
}
